build-scripts/build-rpm.sh

```diff
@@ -15,5 +15,7 @@ rpmbuild \
     --rebuild rpmbuild/SRPMS/*src.rpm
 
 # Move RPMs to exported artifacts
```

Comment:
Can't we just add here single line instead of that complex code below?

```
ARTIFACTS_DIR="${ARTIFACTS_DIR:=$(pwd)/artifacts/rpms/$(date +'%m-%d-%Y')}"
```

Reply:
@mwperina please correct if I am wrong but we only use "pwd/artifacts" if there is no definition of ARTIFACTS_DIR. Are you looking to change that? Also, we need to create the dir if the dir doesn't exist.

I am okay to improving the patch any time.

Reply:
Isn't this

```
ARTIFACTS_DIR="${ARTIFACTS_DIR:=$(pwd)/artifacts/rpms/$(date +'%m-%d-%Y')}"
[[ -d $ARTIFACTS_DIR ]] || mkdir -p $ARTIFACTS_DIR
find rpmbuild -iname \*rpm | xargs mv -t $ARTIFACTS_DIR
```

more simple and clear than

```
ARTIFACTS_DIRECTORY="$(pwd)/artifacts/rpms/$(date +'%m-%d-%Y')"
[[ -z "${ARTIFACTS_DIR}" ]] || ARTIFACTS_DIRECTORY="${ARTIFACTS_DIR}"
[[ -d "${ARTIFACTS_DIRECTORY}" ]] || mkdir -p "${ARTIFACTS_DIRECTORY}"
find rpmbuild -iname \*rpm | xargs mv -t "${ARTIFACTS_DIRECTORY}"
```

?

Reply:
I'd prefer the second since the test for existence of the var (-z) is more explicit - but my script knowledge is also a bit limited.

Reply:
I don't like the 2nd one, it needs additional variable, everything can be cleanly implemented using parameter expansion:

https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html

Reply:
> ARTIFACTS_DIR="${ARTIFACTS_DIR:=$(pwd)/artifacts/rpms/$(date +'%m-%d-%Y')}"
> [[ -d $ARTIFACTS_DIR ]] || mkdir -p $ARTIFACTS_DIR
> find rpmbuild -iname \*rpm | xargs mv -t $ARTIFACTS_DIR

@mwperina  Please correct me, if I am wrong, this is changing the current behaviour of allowing users using env variable to set the artifact directory. IF that's what we are looking, it's fine for me. I can just add this patch instead of the one I sent.